dask/dataframe/multi.py

```diff
@@ -238,11 +238,18 @@ def require(divisions, parts, required=None):
 allowed_right = ("inner", "right")
 
 
-def merge_chunk(lhs, *args, empty_index_dtype=None, categorical_columns=None, **kwargs):
+def merge_chunk(
+    lhs,
+    *args,
+    result_meta,
+    **kwargs,
+):
 
     rhs, *args = args
     left_index = kwargs.get("left_index", False)
     right_index = kwargs.get("right_index", False)
+    empty_index_dtype = result_meta.index.dtype
+    categorical_columns = result_meta.select_dtypes(include="category").columns
```

Comment:
Just to clarify, this isn't a functional change -- it's just cleanup. Instead of computing and passing around `empty_index_dtype` and `categorical_columns` as kwargs, we're passing `result_meta` around and computing `empty_index_dtype` and `categorical_columns` once in the place it's used. Correct?

Reply:
That's right, it's just consolidation of that logic. Now that I'm passing in `result_meta` directly to the chunk function, these derived-from-meta parameters can all just be computed once.